packages/loader/container-loader/src/containerContext.ts

```diff
@@ -157,18 +157,28 @@ export class ContainerContext implements IContainerContext {
         return this._disposed;
     }
 
-    private readonly fluidModuleP = new LazyPromise<IFluidModule>(async () => {
-        const fluidModule = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "CodeLoad" },
+    private readonly fluidModuleP = new LazyPromise<Required<IFluidModuleWithDetails>>(async () => {
+        const loadCodeResult = await PerformanceEvent.timedExecAsync(
+            this.logger,
+            { eventName: "CodeLoad" },
             async () => this.codeLoader.load(this.codeDetails),
         );
 
-        return fluidModule;
+        if ("module" in loadCodeResult) {
+            const { module, details } = loadCodeResult;
+            return {
+                module,
+                details: details ?? this.codeDetails,
+            };
+        } else {
+            return { module: loadCodeResult, details: this.codeDetails };
+        }
     });
 
     constructor(
         private readonly container: Container,
         public readonly scope: IFluidObject,
-        private readonly codeLoader: ICodeLoader,
+        private readonly codeLoader: ICodeDetailsLoader | ICodeLoader,
         public readonly codeDetails: IFluidCodeDetails,
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

i don't think we should expose the incoming value exclusivly. i would make this private, and add a public getter that swaps this for the module code details after load

Reply:
as this is avaialbe to the runtime, and will allow the runtime to access it's own code details

Reply:
In fact, the context doesn't need to keep its own copy of the code details. We'll have the code module with its details instead.

Reply:
I spent some time on the code details dependency between the container and the context. It doesn't seem trivial to me. We probably can revisit it in a follow up work item?

Reply:
can you open an issue for this

Reply:
Done. #6462